tests/rptest/tests/transactions_test.py

```diff
@@ -319,3 +324,113 @@ def on_del(err, msg):
                 prev_rec = bytes(str(int(prev_rec) + 1), 'UTF-8')
 
             num_consumed += len(records)
+
+
+class UpgradeWithMixedVeersionTransactionTest(RedpandaTest):
+    topics = (TopicSpec(partition_count=1, replication_factor=3), )
+
+    def __init__(self, test_context):
+        extra_rp_conf = {
+            "enable_idempotence": True,
+            "enable_transactions": True,
+            "transaction_coordinator_replication": 1,
+            "id_allocator_replication": 1,
+            "enable_leader_balancer": False,
+        }
+
+        super(UpgradeWithMixedVeersionTransactionTest,
+              self).__init__(test_context=test_context,
+                             num_brokers=3,
+                             extra_rp_conf=extra_rp_conf)
+
+        self.installer = self.redpanda._installer
+
+    def on_delivery(self, err, msg):
+        assert err == None, msg
+
+    def check_consume(self, max_records):
+        topic_name = self.topics[0].name
+
+        consumer = ck.Consumer({
+            'bootstrap.servers': self.redpanda.brokers(),
+            'group.id': f"consumer-{uuid.uuid4()}",
+            'auto.offset.reset': 'earliest',
+        })
+
+        consumer.subscribe([topic_name])
+        num_consumed = 0
+        prev_rec = bytes("0", 'UTF-8')
+
+        while num_consumed != max_records:
+            max_consume_records = 10
+            timeout = 10
+            records = consumer.consume(max_consume_records, timeout)
+
+            for record in records:
+                assert prev_rec == record.key(), f"{prev_rec}, {record.key()}"
+                prev_rec = bytes(str(int(prev_rec) + 1), 'UTF-8')
+
+            num_consumed += len(records)
+
+        consumer.close()
+
+    def setUp(self):
+        self.installer.install(self.redpanda.nodes, (22, 1, 5))
```

Comment:
If this is meant to be a general purpose upgrade test for transactions (i.e. not targeting a specific behavioral change), try upgrading from `installer.highest_from_prior_feature_version(RedpandaInstaller.HEAD)` so we get continual coverage from the next latest version of RP.

If this is targeting a specific behavioral change from this PR, would it make sense to upgrade from v22.2?

Reply:
The main idea this test was tested new transaction stuff. I upgraded from 22.1.5, because it uses adl for internal rpc requests